src/SprykerEco/Zed/Payone/Business/Payment/PaymentManager.php

```diff
@@ -1128,7 +1128,7 @@ protected function prepareOrderItems(OrderTransfer $orderTransfer, AbstractReque
 
         foreach ($orderTransfer->getItems() as $itemTransfer) {
             $arrayIt[$key] = PayoneApiConstants::INVOICING_ITEM_TYPE_GOODS;
-            $arrayId[$key] = $itemTransfer->getSku();
+            $arrayId[$key] = substr($itemTransfer->getSku(), 0, 32);
```

Comment:
Instead of silent truncate, wouldnt an exception be better?

Reply:
for developer - yes.
For end-user - no.

I expect here PO to handle this change. in our project we made exactly this change.

Reply:
Well, but what about the case if we have several products, say:
```
1) some-very-amazingly-long-sku-00001
2) some-very-amazingly-long-sku-00002
```
…and they both get trimmed to `some-very-amazingly-long-sku-000`? Will internally `Payone` complain about duplicated `id`?

Reply:
I am also rather in favor of making this visible right now by default.
If a project wants to "just" silence, they can do so by truncating the string beforehand, before it goes into the payment provider, correct?

Reply:
agree with Mark and Darius. 
But if you still want to introduce this logic, you might probable need to do this also in `preparePartialRefundOrderItems` method